internal/plugins/sdk/init.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: rewrite this when plugins phase 2 is implemented.
+package sdk
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+)
+
+// RunInit modifies the project scaffolded by kubebuilder's Init plugin.
+func RunInit(cfg *config.Config) error {
+	// Update the scaffolded Makefile with operator-sdk recipes.
+	if err := initUpdateMakefile("Makefile", cfg); err != nil {
+		return fmt.Errorf("error updating Makefile: %v", err)
+	}
+	return nil
+}
+
+// initUpdateMakefile updates a vanilla kubebuilder Makefile with operator-sdk recipes.
+func initUpdateMakefile(filePath string, cfg *config.Config) error {
+	makefileBytes, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+
+	// Prepend bundle variables.
+	makefileBytes = append([]byte(makefileBundleVarFragment), makefileBytes...)
+
+	// Append bundle recipes.
+	operatorType := projutil.PluginKeyToOperatorType(cfg.Layout)
+	if operatorType == projutil.OperatorTypeGo {
+		makefileBytes = append(makefileBytes, []byte(makefileBundleFragmentGo)...)
+	} else {
+		// if is not go project then, has not the manifest target
+		makefileBytes = append(makefileBytes, []byte(makefileBundleFragmentNonGo)...)
+	}
+
+	makefileBytes = append(makefileBytes, []byte(makefileBundleBuildFragment)...)
+
+	return ioutil.WriteFile(filePath, makefileBytes, 0644)
+}
+
+// Makefile fragments to add to the base Makefile.
+const (
+	makefileBundleVarFragment = `# Current Operator version
+VERSION ?= 0.0.1
+# Default bundle image tag
+BUNDLE_IMG ?= controller-bundle:$(VERSION)
+# Options for 'bundle-build'
+ifneq ($(origin CHANNELS), undefined)
+BUNDLE_CHANNELS := --channels=$(CHANNELS)
+endif
+ifneq ($(origin DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+`
+
```

Comment:
Should we add `.PHONY: bundle`? Otherwise, it seems like users may have to manually delete the bundle directory to get this target to run a second time.

Reply:
Added